Draggability of Now Playing widget

VanishMax opened the pull request:
Closes #132

VanishMax commented:
Ah, forgot about tests... If you are reviewing, please check the logic and code first

illright commented on src/lib/widgets/now-playing/ui/episode-display.svelte:
```diff
@@ -4,7 +4,10 @@
 </script>
 
-<img class="flex rounded-2xl" src={podcast.coverUrl} alt="" />
+<div
```
question: why is this necessary?

VanishMax replied:
There was a problem of image always taking the full width of the container and scaling vertically to enormous sizes. I wanted to try solving it with `img` tag but failed, so the `background` came up to place perfectly.

VanishMax commented:
@alkaitagi @illright You might want to check this PR again – I've added tests and refactored the code, since the were many changes from the `main` branch

illright commented on src/lib/widgets/now-playing/lib/detect-swipe.ts:
```diff
@@ -0,0 +1,66 @@
+
+type SwipeCallbacks = Partial<Record<'left' | 'right' | 'up' | 'down', () => void>>;
+
+export const detectSwipe = (callbacks: SwipeCallbacks) => {
```
suggestion (non-blocking): would be even cooler to make this a Svelte action, something like this:
```svelte
<div use:swipe={{ down: () => dispatch('minimize') }} />
```

illright commented on src/lib/widgets/now-playing/ui/swipe-down-bar.svelte:
```diff
@@ -0,0 +1,21 @@
+  });
+</script>
+
+<button
```
question: why is it a button? Looks like this will produce very confusing screen reader semantics

illright commented on src/lib/widgets/now-playing/ui/now-playing.svelte:
```diff
@@ -54,6 +55,7 @@
           />
         {/if}
         <div class="flex flex-col items-center" class:invisible={transcriptShown}>
+          <SwipeDownBar on:minimize={() => nowPlayingActive.set(false)} />
```
issue: I think we still want to show the drag indicator even when the transcript is shown

illright commented on src/lib/widgets/now-playing/specs/swipe-down-bar.spec.ts:
```diff
@@ -0,0 +1,87 @@
+   */
+  fireEvent(
+    screen.getByRole('button'),
+    Object.assign(
```
praise: this is some intense logic there, and I appreciate the research

issue: However, from what I discovered about JSDOM and the Testing Library is that testing drag is not a good idea in JSDOM, since it doesn't have layout at all. Maintainers of Testing Library suggest using Cypress for testing such stuff since it requires a real browser.

suggestion: perhaps we shouldn't cover this particular component with unit tests from Jest?

References:
- https://github.com/testing-library/user-event/issues/440#issuecomment-685010755 (This Kent C. Dodds guy is probably the most respectable guy in terms of testing from my current perspective)
- https://github.com/testing-library/user-event/issues/871 (An issue I created for the Testing Library when I was testing the scrubbing bar)

VanishMax replied:
Yeah, I've read these too, but couldn't understand what "no layout" actually mean. I'd appreciate you explaining the concept, and will easily remove the test

illright replied:
Well, in simple terms, there's no "coordinate system", elements don't actually have a width and height. Everything is just collapsed into a single dot. That's how I understand it, anyway

illright replied:
Which is why writing code that involves distances is risky and flaky in JSDOM

VanishMax replied:
Removed the test

illright reviewed (changes requested):
Sorry for not mentioning it earlier, but this is actually a little more involved of a task. According to the acceptance criteria, the panel should follow your finger as you drag it around, not just minimize on a swipe-down gesture.
